### ISSUE TITLE
NPE in MediaCodecAudioRenderer

### ISSUE THREAD

**Issue by stari4ek** [google/ExoPlayer#6674]
### [REQUIRED] Issue description
I'm seeing small amount of NullPointerException in crashlytics related to ExoPlayer.

### [REQUIRED] Reproduction steps
it's reported automatically, I do not own stream source or device to reproduce

### [REQUIRED] Link to test content
it's reported automatically, I do not own stream source or device to reproduce

### [REQUIRED] A full bug report captured from the device
Here is exoplayer log with partially deobfuscated callstack.
Checking the source code, there are multiple code paths where `MimeTypes.getEncoding` can be called with null from `MediaCodecAudioRenderer`.
```
323 18:13:28.989|main|D] EventLogger: decoderDisabled [eventTime=101.62, mediaPos=0.00, window=0, period=0, video]
323 18:13:28.994|main|D] EventLogger: decoderDisabled [eventTime=101.63, mediaPos=0.00, window=0, period=0, audio]
323 18:13:29.076|main|D] EventLogger: mediaPeriodReleased [eventTime=101.71, mediaPos=0.00, window=0, period=0]
323 18:13:29.078|main|D] BandwidthLogger: Bandwidth stats: elapsed: 408 ms, data: 1 MB, bitrate: 21Mbit/s
323 18:13:29.086|main|E] EventLogger: playerFailed [eventTime=101.72, mediaPos=0.00, window=0, period=0]
com.google.android.exoplayer2.ExoPlaybackException: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:24)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:193)
	at android.os.HandlerThread.run(HandlerThread.java:65)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
	at com.google.android.exoplayer2.util.MimeTypes.getEncoding(MimeTypes.java:1)
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.onAudioSessionId(MediaCodecAudioRenderer.java:4)
                                                                getPassthroughEncoding
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.onAudioTrackUnderrun(MediaCodecAudioRenderer.java:31)
                                                                allowPassthrough
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.onAudioTrackUnderrun(MediaCodecAudioRenderer.java:20)
                                                                getDecoderInfos
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.onQueueInputBuffer(MediaCodecRenderer.java:2)
                                                                getAvailableCodecInfos
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.canKeepCodec(MediaCodecRenderer.java:31)
                                                                configureCodec
                                                                getCodecOperatingRateV23
                                                                getDecoderInfos
                                                                handleInputBufferSupplementalData
                                                                onCodecInitialized
                                                                onOutputFormatChanged
                                                                processOutputBuffer
                                                                shouldInitCodec
                                                                supportsFormat
                                                                maybeInitCodecWithFallback
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodec(MediaCodecRenderer.java:15)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.reinitializeCodec(MediaCodecRenderer.java:2)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.drainAndReinitializeCodec(MediaCodecRenderer.java:4)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.canKeepCodec(MediaCodecRenderer.java:139)
                                                                configureCodec
                                                                getCodecOperatingRateV23
                                                                getDecoderInfos
                                                                handleInputBufferSupplementalData
                                                                onCodecInitialized
                                                                onOutputFormatChanged
                                                                processOutputBuffer
                                                                shouldInitCodec
                                                                supportsFormat
                                                                onInputFormatChanged
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.onAudioTrackUnderrun(MediaCodecAudioRenderer.java:51)
                                                                onInputFormatChanged
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.feedInputBuffer(MediaCodecRenderer.java:32)
```

### [REQUIRED] Version of ExoPlayer being used
d874656e8a7668a83c6702413f094f6a1f509070

### [REQUIRED] Device(s) and version(s) of Android being used
* Xiaomi MIBOX4
* OS Level 28

**Comment by andrewlewis** [comment]
The stack trace doesn't make sense (e.g., `MediaCodecAudioRenderer.onAudioTrackUnderrun` doesn't call anything). Please check you're using the right map for deobfuscation.

**Comment by stari4ek** [comment]
I'm using the proper mapping file. When there are multiple mappings for same name (but with different arguments resolution) it produces this kind of deobfuscated stack trace with variants.

Here is manually unrolled callstack going from top-to-bottom, checking who-calls-who:

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
	at com.google.android.exoplayer2.util.MimeTypes.getEncoding(MimeTypes.java:1)
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.getPassthroughEncoding()
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.allowPassthrough()
	at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.getDecoderInfos()
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.getAvailableCodecInfos()
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecWithFallback()
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodec(MediaCodecRenderer.java:15)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.reinitializeCodec(MediaCodecRenderer.java:2)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.drainAndReinitializeCodec(MediaCodecRenderer.java:4)
```

btw, If you enable lint on `MediaCodecAudioRenderer.java` - there're bunch of Null-warnings related to `getPassthroughEncoding`

**Comment by andrewlewis** [comment]
Ah I see. Thanks for the clarification. I think the `NullPointerException` is already fixed in `dev-v2` by aeefb79a86519a1539f503fb5ac2b3f0c7760ba7, but I wonder if there's something else going wrong here that causes the MIME type to be null. On the development branch you may just end up getting `DecoderInitializationException` instead.

**Comment by stari4ek** [comment]
Yeah, this commit looks like the better way to handle it.
Will bump to current HEAD from dev-v2.

Thank you
